crates/wasmparser/src/validator/types.rs

```diff
@@ -465,7 +465,7 @@ define_wrapper_id! {
     pub enum ComponentCoreTypeId {
         #[unwrap = unwrap_sub]
         /// A core type.
-        Sub(CoreTypeId),
+        Rec(CoreTypeId),
```

Comment:
This'll actually stay as `Sub` unlike some of the other refactorings I've since realized. (it's not quite obvious given the current structure of the code but https://github.com/bytecodealliance/wasm-tools/pull/1765 should make it more clear as to why)